### PR TITLE
create: add --crystal option

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Enable easy creation of formula for tools implemented in Crystal.

### CODE REVIEW

File: Library/Homebrew/formula_creator.rb [hunk 1]
```diff
@@ -138,6 +140,9 @@ def install
                                   "--disable-dependency-tracking",
                                   "--disable-silent-rules",
                                   "--prefix=\#{prefix}"
+        <% elsif mode == :crystal %>
+            system "shards", "build", "--release"
+            bin.install "bin/#{name}"
```
Comment (dawidd6) [inline Library/Homebrew/formula_creator.rb:145]:
```suggestion
            bin.install bin/name
```

**Comment by MikeMcQuaid** [comment]
@vladimyr can you email me at mike@mikemcquaid.com (so I have your address)? Thanks!

File: Library/Homebrew/formula_creator.rb (hunk 1 above)
Comment (vladimyr) [reply Library/Homebrew/formula_creator.rb:145]: Yeah, it makes sense. I guess there is million ways to do things in Ruby and I have tendency to constantly pick the wrong one 🤣

Comment (vladimyr) [reply Library/Homebrew/formula_creator.rb]: @dawidd6 Wait a minute, this references bin dir inside prefix while compilation outputs are stored inside bin dir under build path? Am I missing something? 🤔

Comment (dawidd6) [reply Library/Homebrew/formula_creator.rb]: Oh man, sorry. I'm wrong on this one, you are right!

Comment (vladimyr) [reply Library/Homebrew/formula_creator.rb:145]: No worries 👍

**Comment by vladimyr** [comment]
> @vladimyr can you email me at [mike@mikemcquaid.com](mailto:mike@mikemcquaid.com) (so I have your address)? Thanks!

Sent you an email ✅

**Pull request merged**